components/script/dom/htmlformelement.rs

```diff
@@ -417,55 +412,77 @@ impl HTMLFormElement {
 
         let mut load_data = LoadData::new(action_components, doc.get_referrer_policy(), Some(doc.url().clone()));
 
-        let parsed_data = match enctype {
-            FormEncType::UrlEncoded => {
-                load_data.headers.set(ContentType::form_url_encoded());
-
-                form_urlencoded::Serializer::new(String::new())
-                    .encoding_override(Some(self.pick_encoding()))
-                    .extend_pairs(form_data.into_iter().map(|field| (field.name.clone(), field.value_str())))
-                    .finish()
-            }
-            FormEncType::FormDataEncoded => {
-                let boundary = self.generate_boundary();
-                let mime = mime!(Multipart / FormData; Boundary =(&boundary));
-                load_data.headers.set(ContentType(mime));
-
-                self.encode_multipart_form_data(&mut form_data, None, boundary)
-            }
-            FormEncType::TextPlainEncoded => {
-                load_data.headers.set(ContentType(mime!(Text / Plain)));
-
-                self.encode_plaintext(&mut form_data)
-            }
-        };
```

Comment:
This can't be shared as I see, since we will put data in two places depending on the method and scheme:

1. Request URL (`mutate_action_url`): It needs string-encoded data, even for file upload (but certainly we don't put file content bytes into URL query string, but just file names iirc)
2. Request body (`submit_entity_body`): It *doesn't* need to encode whole content as string, esp. when source is file upload, it will simply append bytes to the body.

So, the `String` type `parsed_data` can't be used in body. And as spec says, `mutate_entity_url` has nothing to do with `FormEncType` when composing the query string.